src/cli/ofp_cli.c

```diff
@@ -186,6 +186,35 @@ static int ip4net_ok(char *val)
 	return 1;
 }
 
+static int mac_ok(char *val)
+{
+	char *tk_start, *tk_end, *val_end, *ch;
+	int tk_cnt = 0;
+
+	for(tk_cnt = 0, tk_start = val, val_end = val + strlen(val);
+		tk_start != val_end;
+		tk_cnt++, tk_start = (*tk_end == '\0')? tk_end: tk_end + 1)
```

Comment:
This for loop is quite hard to understand and could probably be simplified a bit.